### ISSUE TITLE
how to track custom events using Watson?

### ISSUE THREAD

**Issue by pgoldweic** [watson-developer-cloud/botkit-middleware#136]
I would like to add to my Watson dialog the capability to indicate that a certain custom event should be tracked (for example, to do some custom analytics later on) given a particular user response within the dialog (e.g. only some of the 'yes' responses by the bot user should be tracked as 'helpful' responses, so I might want to emit/track a 'helpful response' event given these specific 'yes' responses). How would I do this? Is there a best practice? I imagine that I might need to add a new field to the message object that would be available later on to some other middleware (which would in turn emit this custom event to some server), but I'm not sure how to do this. Any info will be appreciated.

**Comment by pgoldweic** [comment]
I've managed to apparently add a field to the Watson response (via JSON editor in Watson Assistant) that carries info. about a custom event. Then I've been able to access this information via this middleware by referring to: message.WatsonData.output.<my-field-name>, and I've decided to use this to implement this custom event tracking. Any comments on this, or info. about best practices?

**Comment by pgoldweic** [comment]
For the record in case it helps somebody else: I ended up realizing that this option was not really a good one after all, since the additional field I was adding to the watsonData.output property was *not* something that persisted in cases where the dialog node in which the field was added to was an 'internal' dialog node (by this I mean a dialog node that is processed and may produce response text but it is not 'final' -in the sense that it may cause execution to jump to other nodes which are still going to be contributing to the response offered to the user). 
My solution was to create an array context variable that keeps track of all the custom events I need to track (this array is appended to as Watson executes a complete processing path). After processing the Watson response, my bot application then tracks all these events, clears the array and continues.
